app/src/main/java/net/bible/service/db/readingplan/ReadingPlanDatabaseDefinition.kt

```diff
@@ -0,0 +1,246 @@
+ *
+ */
+
+/**
+ *
+ */
+package net.bible.service.db.readingplan
+
+import android.content.ContentValues
+import android.database.sqlite.SQLiteDatabase
+import android.provider.BaseColumns
+import android.util.Log
+import net.bible.service.common.CommonUtils
+
+import net.bible.service.readingplan.ReadingPlanDao
+import net.bible.service.readingplan.ReadingPlanInfoDto
+
+/**
+ * @author Timmy Braun [tim.bze at gmail dot com] (2/12/2019)
+ * TODO: Figure out something that a plan like Prof Horner 10 chapter a day,
+ * which is 10 groups of chapters, will continue to loop through the groups
+ * even after the year is up
+ */
+object ReadingPlanDatabaseDefinition {
+    private const val TAG = "ReadingPlanDatabaseDef"
+    const val DB_TRUE_VALUE = "1"
+    const val DB_FALSE_VALUE = "0"
+
+    object ReadingPlanMeta : BaseColumns {
+        // If COLUMN_CURRENT_DAY has CONSTANT_CURRENT_DAY_BY_DATE value,
+        // then the reading plan will by default get today's date
+        const val CONSTANT_CURRENT_DAY_BY_DATE = -1
+
+        const val TABLE_NAME = "readingplan_meta"
+        const val COLUMN_ID = BaseColumns._ID
+        const val COLUMN_PLAN_FILE_NAME = "plan_file_name"
+        const val COLUMN_PLAN_NAME = "plan_name"
+        const val COLUMN_DESCRIPTION = "description"
+        const val COLUMN_DATE_START = "date_start"
+        const val COLUMN_DAYS_IN_PLAN = "days_in_plan"
+        const val COLUMN_CURRENT_DAY = "current_day"
+        const val COLUMN_VERSIFICATION_NAME = "versification"
+        const val COLUMN_LAST_USED_PLAN = "last_used"
+    }
+
+    object ReadingPlanDays : BaseColumns {
+        const val TABLE_NAME = "readingplan_days"
+        const val COLUMN_ID = BaseColumns._ID
+        const val COLUMN_READING_PLAN_META_ID = "readingplan_meta_id"
+        const val COLUMN_DAY_NUMBER = "day_number"
+        const val COLUMN_READING_DATE = "reading_date"
+        const val COLUMN_DAY_CHAPTERS = "day_chapters"
+        const val COLUMN_READ_STATUS = "read_status"
+    }
+
+    class Operations {
+        companion object {
+            private const val TAG = "ReadingPlanDatabaseOps"
+            private const val SQL_CREATE_ENTRIES_META =
+                "CREATE TABLE ${ReadingPlanMeta.TABLE_NAME} (" +
+                "${ReadingPlanMeta.COLUMN_ID} INTEGER PRIMARY KEY AUTOINCREMENT," +
+                "${ReadingPlanMeta.COLUMN_PLAN_FILE_NAME} TEXT NOT NULL," +
+                "${ReadingPlanMeta.COLUMN_PLAN_NAME} TEXT NOT NULL," +
+                "${ReadingPlanMeta.COLUMN_DESCRIPTION} TEXT NOT NULL," +
+                "${ReadingPlanMeta.COLUMN_DATE_START} INTEGER," +
+                "${ReadingPlanMeta.COLUMN_DAYS_IN_PLAN} INTEGER," +
+                "${ReadingPlanMeta.COLUMN_CURRENT_DAY} INTEGER NOT NULL DEFAULT 0," +
+                "${ReadingPlanMeta.COLUMN_VERSIFICATION_NAME} TEXT NOT NULL," +
+                "${ReadingPlanMeta.COLUMN_LAST_USED_PLAN} INTEGER NOT NULL DEFAULT 0" +
+                "); "
+
```

Comment:
Please use Kotlin multiline string literals  (`"""..."""`)

Reply:
Here and in other similar places

Reply:
> If they are used in multiple classes and are not strictly semantically connected to the particular class, they might be module level variables too.

Can you give me an example of what you mean with module level variable? I do not really want to put this line more than once, since it has to do with the specific way the date-based plan needs to have the date specified: SimpleDateFormat("MMM-d/yyyy") -- I am also using it in OneDaysReadingsDto.

Reply:
> Essentially you are wanting to store which reading plan is active, right? Then I think you should write somewhere just the index of that entry, instead of boolean value on each column.

Yes, would you make another table just for that? I don't see how I could do that in the 2 tables I'm using now.

Reply:
Until then maybe this could be stored in current shared preferences.